src/main/java/com/royalrangers/service/UserService.java

```diff
@@ -154,5 +159,101 @@ public UserAgeGroup determineUserAgeGroup(int userAge) {
         return listUsersBeanToApprove;
     }
 
+    public boolean hasRole(String role) {
+        Collection<GrantedAuthority> authorities = (Collection<GrantedAuthority>)
+                SecurityContextHolder.getContext().getAuthentication().getAuthorities();
+        boolean hasRole = false;
+
+        for (GrantedAuthority authority : authorities) {
+            hasRole = authority.getAuthority().equals(role);
+            if (hasRole) {
+                break;
+            }
+        }
+        return hasRole;
+    }
+
+    public String getLoggedUserEmail() {
+        return SecurityContextHolder.getContext().getAuthentication().getName();
+    }
+
+    public void updateUserByEmail(String email, UserBean update) {
+        if(!hasRole("ROLE_ADMIN")) {
+            shrinkProperties(update);
+        }
+        User user = userRepository.findByEmail(email);
+        updateUser(user, update);
+    }
+
+    private void shrinkProperties(UserBean update) {
+        update.setFirstName(null);
+        update.setLastName(null);
+        update.setGender(null);
+        update.setUserAgeGroup(null);
+        update.setGroupId(null);
+        update.setSectionId(null);
+        update.setPlatoonId(null);
+    }
+
+    public void updateUserById(Long id, UserBean update) {
+        if(!userRepository.exists(id)) {
+            throw new UserRepositoryException("Not found user with id " + id);
+        }
+        User user = userRepository.findOne(id);
+        updateUser(user, update);
+    }
+
+    public void updateUser(User user, UserBean update) {
+
+        if (update.getFirstName() != null) {
```

Comment:
Please explaine why do you need all of this checks into this method. From the UI you'll get full filled user object with updated information. So you don't need all of this checks. Just update all user fields and save user entity.